src/java.base/share/classes/java/lang/runtime/Carriers.java

```diff
@@ -0,0 +1,1005 @@
+                component = MethodHandles.filterReturnValue(
+                        components[intIndex++], INT_TO_FLOAT);
+            } else if (ptype == long.class) {
+                component = components[longIndex++];
+            } else {
+                component = components[intIndex++];
+            }
+
+            MethodType methodType = methodType(ptype, CarrierObject.class);
+            reorder[index++] =
+                    MethodHandles.explicitCastArguments(component, methodType);
+        }
+
+        return List.of(reorder);
+    }
+
+    /**
+     * Factory for carriers that are backed by long[] and Object[].
+     */
+    static final class CarrierObjectFactory {
+        /**
+         * Unsafe access.
+         */
+        private static final Unsafe UNSAFE;
+
+        /*
+         * Constructor accessor MethodHandles.
+         */
+        private static final MethodHandle CONSTRUCTOR;
+        private static final MethodHandle GET_LONG;
+        private static final MethodHandle PUT_LONG;
+        private static final MethodHandle GET_INTEGER;
+        private static final MethodHandle PUT_INTEGER;
+        private static final MethodHandle GET_OBJECT;
+        private static final MethodHandle PUT_OBJECT;
+
+        static {
+            try {
+                UNSAFE = Unsafe.getUnsafe();
+                Lookup lookup = MethodHandles.lookup();
+                CONSTRUCTOR = lookup.findConstructor(CarrierObject.class,
+                        methodType(void.class, int.class, int.class));
+                GET_LONG = lookup.findVirtual(CarrierObject.class, "getLong",
+                        methodType(long.class, int.class));
+                PUT_LONG = lookup.findVirtual(CarrierObject.class, "putLong",
+                        methodType(CarrierObject.class, int.class, long.class));
+                GET_INTEGER = lookup.findVirtual(CarrierObject.class, "getInteger",
+                        methodType(int.class, int.class));
+                PUT_INTEGER = lookup.findVirtual(CarrierObject.class, "putInteger",
+                        methodType(CarrierObject.class, int.class, int.class));
+                GET_OBJECT = lookup.findVirtual(CarrierObject.class, "getObject",
+                        methodType(Object.class, int.class));
+                PUT_OBJECT = lookup.findVirtual(CarrierObject.class, "putObject",
+                        methodType(CarrierObject.class, int.class, Object.class));
+            } catch (ReflectiveOperationException ex) {
+                throw new AssertionError("carrier static init fail", ex);
+            }
+        }
+
+        /**
+         * Constructor builder.
+         *
+         * @param carrierShape  carrier object shape
+         *
+         * @return {@link MethodHandle} to generic carrier constructor.
+         */
+        MethodHandle constructor(CarrierShape carrierShape) {
+            int longCount = carrierShape.longCount();
```

Comment:
`longCount`/`intCount` seems unused in this method

Reply:
Changing